tests/core/test_integration.py

```diff
@@ -4494,6 +4496,128 @@ def test_multi(mocker):
     ]
 
 
+@use_terminal_console
+def test_multi_virtual_layer(mocker):
+    context = Context(paths=["tests/fixtures/multi_virtual_layer"])
+
+    local_db = "db.duckdb"
+    if os.path.exists(local_db):
```

Comment:
To avoid problems with this state as well as concurrent test runs I suggest copying the project folder into a temporary location (use the `tmp_path` fixture)